packages/test/test-end-to-end-tests/src/test/idCompressor.spec.ts

```diff
@@ -1084,3 +1085,91 @@ describeCompat("IdCompressor Summaries", "NoCompat", (getTestObjectProvider, com
 		assert(dds2.id.length > 8, "DDS's ID in attached container should not be short");
 	});
 });
+
+const dataStoreCount = 13;
+describeCompat("IdCompressor basic", "NoCompat", (getTestObjectProvider, apis) => {
+	const {
+		dataRuntime: { DataObject, DataObjectFactory },
+		containerRuntime: { ContainerRuntimeFactoryWithDefaultDataStore },
+	} = apis;
+	class TestDataObject extends DataObject {
+		public get _root() {
+			return this.root;
+		}
+
+		public get _context() {
+			return this.context;
+		}
+	}
+
+	const configProvider = createTestConfigProvider();
+	let provider: ITestObjectProvider;
+	const defaultFactory = new DataObjectFactory("T", TestDataObject, [], []);
```

Comment:
You can simply create the following and then you don't need the `TestDataObject` either:
`const defaultFactory = new TestFluidObjectFactory([]);`

Reply:
I need the `DataObjectFactory` so that I can create `TestDataObject`.

Reply:
The test isn't doing anything special with it. You could create new data stores via `containerRuntime.createDataStore(defaultFactory.type)` instead.
Anyway, just a suggestion to make the amount of code smaller :-)

Reply:
I think I'm going to choose to stick with the `DataObjectFactory` pattern because I think that's the usage pattern we want our customers to go towards. If we in the future try to hide the `containerRuntime`, this will be one less test to fix.